File: e2e/go/map_test.go
// E2e tests for category: map
package e2e_test

import (
	"os"
	"testing"

	pkg "github.com/kreuzberg-dev/kreuzcrawl"
)

func Test_MapDiscoverUrls(t *testing.T) {
	// Discovers all URLs on a site without fetching full content
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	url := os.Getenv("MOCK_SERVER_URL") + "/fixtures/map_discover_urls"
	_, err := pkg.Scrape(engine, url)
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'urls.length' not available on result type
}

func Test_MapExcludePatterns(t *testing.T) {
	// Excludes URLs matching patterns from URL map
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	url := os.Getenv("MOCK_SERVER_URL") + "/fixtures/map_exclude_patterns"
	_, err := pkg.Scrape(engine, url)
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'urls.length' not available on result type
}

func Test_MapIncludeSubdomains(t *testing.T) {
	// Includes subdomain URLs in URL map discovery
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	url := os.Getenv("MOCK_SERVER_URL") + "/fixtures/map_include_subdomains"
	_, err := pkg.Scrape(engine, url)
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'urls.length' not available on result type
	// skipped: field 'urls' not available on result type
}

func Test_MapLargeSitemap(t *testing.T) {
	// Handles large sitemap with 100+ URLs
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	url := os.Getenv("MOCK_SERVER_URL") + "/fixtures/map_large_sitemap"
	_, err := pkg.Scrape(engine, url)
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'urls.length' not available on result type
}

func Test_MapLimitPagination(t *testing.T) {
	// Limits map result count to specified maximum
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	url := os.Getenv("MOCK_SERVER_URL") + "/fixtures/map_limit_pagination"
	_, err := pkg.Scrape(engine, url)
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'urls.length' not available on result type
}

func Test_MapSearchFilter(t *testing.T) {
	// Filters map results by search keyword
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	url := os.Getenv("MOCK_SERVER_URL") + "/fixtures/map_search_filter"
	_, err := pkg.Scrape(engine, url)
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'urls.length' not available on result type
	// skipped: field 'urls' not available on result type
}

File: e2e/java/src/test/java/dev/kreuzberg/e2e/MapTest.java
package dev.kreuzberg.e2e;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/** E2e tests for category: map. */
class MapTest {
    @Test
    void testMapDiscoverUrls() throws Exception {
        // Discovers all URLs on a site without fetching full content
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        assertTrue(result.urls().size() >= 3, "expected >= 3");
    }

    @Test
    void testMapExcludePatterns() throws Exception {
        // Excludes URLs matching patterns from URL map
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        assertEquals(1, result.urls().size());
    }

    @Test
    void testMapIncludeSubdomains() throws Exception {
        // Includes subdomain URLs in URL map discovery
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        assertTrue(result.urls().size() >= 2, "expected >= 2");
        assertTrue(result.urls().contains("blog.example.com"), "expected to contain: " + "blog.example.com");
    }

    @Test
    void testMapLargeSitemap() throws Exception {
        // Handles large sitemap with 100+ URLs
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        assertTrue(result.urls().size() >= 100, "expected >= 100");
    }

    @Test
    void testMapLimitPagination() throws Exception {
        // Limits map result count to specified maximum
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        assertTrue(result.urls().size() <= 5, "expected <= 5");
    }

    @Test
    void testMapSearchFilter() throws Exception {
        // Filters map results by search keyword
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        assertTrue(result.urls().size() >= 2, "expected >= 2");
        assertTrue(result.urls().contains("blog"), "expected to contain: " + "blog");
    }

}
